### PR TITLE
New() goes in the trash, we've gotta Initialize()

### PR DESCRIPTION
**Pull request by ChangelingRain**
something something initialize

### CODE REVIEW

**Comment by optimumtact** [comment]
>"While we normally encourage (and in some cases, even require) bringing out of date code up to date when you make unrelated changes near the out of date code, that is not the case for New -> Initialize conversions. These systems are generally more dependant on parent and children procs so unrelated random conversions of existing things can cause bugs that take months to figure out."

**Comment by Jalleo** [comment]
Reduction of lines?!?!?

**Comment by ChangelingRain** [comment]
>when you make unrelated changes near the out of date code

which is not the case here because hey wow this is specifically FOR it

**Comment by Cyberboss** [comment]
Problems with Initialize only exist if they have New()s above or below them in the inheritance hierarchy. This is fine

File: code/game/gamemodes/miniantags/abduction/machinery/pad.dm [hunk 1]
```diff
@@ -47,8 +47,8 @@
 	icon_state = "teleport"
 	duration = 80
 
-/obj/effect/overlay/temp/teleport_abductor/New()
+/obj/effect/overlay/temp/teleport_abductor/Initialize()
+	. = ..()
 	var/datum/effect_system/spark_spread/S = new
 	S.set_up(10,0,loc)
-	S.start()
-	..()
```
Comment (MrStonedOne) [inline code/game/gamemodes/miniantags/abduction/machinery/pad.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/game/objects/effects/overlays.dm [hunk 2]
```diff
@@ -45,10 +45,10 @@
 /obj/effect/overlay/temp/dir_setting
 	randomdir = FALSE
 
-/obj/effect/overlay/temp/dir_setting/New(loc, set_dir)
+/obj/effect/overlay/temp/dir_setting/Initialize(mapload, set_dir)
 	if(set_dir)
 		setDir(set_dir)
-	..()
```
Comment (MrStonedOne) [inline code/game/objects/effects/overlays.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/game/objects/effects/overlays.dm [hunk 3]
```diff
@@ -57,12 +57,12 @@
 	layer = BELOW_MOB_LAYER
 	var/splatter_type = "splatter"
 
-/obj/effect/overlay/temp/dir_setting/bloodsplatter/New(loc, set_dir)
+/obj/effect/overlay/temp/dir_setting/bloodsplatter/Initialize(mapload, set_dir)
 	if(set_dir in GLOB.diagonals)
 		icon_state = "[splatter_type][pick(1, 2, 6)]"
 	else
 		icon_state = "[splatter_type][pick(3, 4, 5)]"
-	..()
```
Comment (MrStonedOne) [inline code/game/objects/effects/overlays.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/game/objects/effects/overlays.dm [hunk 4]
```diff
@@ -325,15 +332,15 @@
 	var/mob/user
 	var/damage = 20
 
-/obj/effect/overlay/temp/ratvar/volt_hit/New(loc, caster, multiplier)
+/obj/effect/overlay/temp/ratvar/volt_hit/Initialize(mapload, caster, multiplier)
 	if(multiplier)
 		damage *= multiplier
 	duration = max(round(damage * 0.2), 1)
-	..()
```
Comment (MrStonedOne) [inline code/game/objects/effects/overlays.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/game/objects/effects/overlays.dm [hunk 5]
```diff
@@ -494,9 +496,9 @@
 	icon = 'icons/mob/mob.dmi'
 	duration = 15
 
-/obj/effect/overlay/temp/gib_animation/New(loc, gib_icon)
+/obj/effect/overlay/temp/gib_animation/Initialize(mapload, gib_icon)
 	icon_state = gib_icon // Needs to be before ..() so icon is correct
-	..()
```
Comment (MrStonedOne) [inline code/game/objects/effects/overlays.dm]: See comment

This was running code before parent New() that would now happen after New().

Would the behavior be correct now that its running that code after new but before initialize?

File: code/game/objects/effects/overlays.dm [hunk 6]
```diff
@@ -508,9 +510,9 @@
 	icon = 'icons/mob/mob.dmi'
 	duration = 15
 
-/obj/effect/overlay/temp/dust_animation/New(loc, dust_icon)
+/obj/effect/overlay/temp/dust_animation/Initialize(mapload, dust_icon)
 	icon_state = dust_icon // Before ..() so the correct icon is flick()'d
-	..()
```
Comment (MrStonedOne) [inline code/game/objects/effects/overlays.dm]: See comment

This was running code before parent New() that would now happen after New().

Would the behavior be correct now that its running that code after new but before initialize?

File: code/game/objects/effects/overlays.dm [hunk 7]
```diff
@@ -559,14 +561,14 @@
...
 
-/obj/effect/overlay/temp/impact_effect/New(loc, atom/target, obj/item/projectile/P)
+/obj/effect/overlay/temp/impact_effect/Initialize(mapload, atom/target, obj/item/projectile/P)
 	if(target == P.original) //the projectile hit the target originally clicked
 		pixel_x = P.p_x + target.pixel_x - 16 + rand(-4,4)
 		pixel_y = P.p_y + target.pixel_y - 16 + rand(-4,4)
 	else
 		pixel_x = target.pixel_x + rand(-4,4)
 		pixel_y = target.pixel_y + rand(-4,4)
-	..()
```
Comment (MrStonedOne) [inline code/game/objects/effects/overlays.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/modules/projectiles/projectile/special.dm [hunk 8]
```diff
@@ -215,7 +216,6 @@
 			name = "full strength [name]"
 			damage *= 4
 			range *= 2
-	..()
```
Comment (MrStonedOne) [inline code/modules/projectiles/projectile/special.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 9]
```diff
@@ -17,12 +17,12 @@
 	var/useramount = 30 // Last used amount
 	layer = BELOW_OBJ_LAYER
 
-/obj/machinery/chem_master/New()
+/obj/machinery/chem_master/Initialize()
 	create_reagents(100)
 	add_overlay("waitlight")
 	var/obj/item/weapon/circuitboard/machine/B = new /obj/item/weapon/circuitboard/machine/chem_master(null)
 	B.apply_default_parts(src)
-	..()
```
Comment (MrStonedOne) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/modules/reagents/reagent_containers/blood_pack.dm [hunk 10]
```diff
@@ -43,9 +43,9 @@
 		if(51 to INFINITY)
 			icon_state = "full"
 
-/obj/item/weapon/reagent_containers/blood/random/New()
+/obj/item/weapon/reagent_containers/blood/random/Initialize()
 	blood_type = pick("A+", "A-", "B+", "B-", "O+", "O-", "L")
-	..()
```
Comment (MrStonedOne) [inline code/modules/reagents/reagent_containers/blood_pack.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/modules/reagents/reagent_dispenser.dm [hunk 11]
```diff
@@ -23,10 +23,10 @@
 	else
 		return ..()
 
-/obj/structure/reagent_dispensers/New()
+/obj/structure/reagent_dispensers/Initialize()
 	create_reagents(tank_volume)
 	reagents.add_reagent(reagent_id, tank_volume)
-	..()
```
Comment (MrStonedOne) [inline code/modules/reagents/reagent_dispenser.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/modules/research/message_server.dm [hunk 12]
```diff
@@ -78,12 +78,11 @@ GLOBAL_LIST_INIT(message_servers, list())
 	var/active = 1
 	var/decryptkey = "password"
 
-/obj/machinery/message_server/New()
+/obj/machinery/message_server/Initialize()
 	GLOB.message_servers += src
 	decryptkey = GenerateKey()
 	send_pda_message("System Administrator", "system", "This is an automated message. The messaging system is functioning correctly.")
-	..()
-	return
```
Comment (MrStonedOne) [inline code/modules/research/message_server.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/modules/shuttle/emergency.dm [hunk 13]
```diff
@@ -533,13 +532,13 @@
...
 	roundstart_move = "backup_away"
 
-/obj/docking_port/mobile/emergency/backup/New()
+/obj/docking_port/mobile/emergency/backup/Initialize()
 	// We want to be a valid emergency shuttle
 	// but not be the main one, keep whatever's set
 	// valid.
 	// backup shuttle ignores `timid` because THERE SHOULD BE NO TOUCHING IT
 	var/current_emergency = SSshuttle.emergency
-	..()
```
Comment (MrStonedOne) [inline code/modules/shuttle/emergency.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

File: code/modules/spells/spell_types/godhand.dm [hunk 14]
```diff
@@ -14,9 +14,9 @@
 	throw_range = 0
 	throw_speed = 0
 
-/obj/item/weapon/melee/touch_attack/New(var/spell)
-	attached_spell = spell
-	..()
```
Comment (MrStonedOne) [inline code/modules/spells/spell_types/godhand.dm]: This was running code before parent New() that would now happen after New().

Does this matter in this case?

**Review by MrsTonedOne — COMMENTED** [review]
Ok, So i know this is basically the same fucking comment repeated 14 million times, but after a New -> Initialize caused a full server lockup. I figured I'd be thorough.

Blame @optimumtact for requesting my review.

edit: also, i'm pretty sure you already thought of these things and checked them, so don't take this as patronizing, I just want confirmation. =P

**Comment by ChangelingRain** [comment]
I'm not actually going to reply to each comment, but yes I did check all of those.

**Pull request merged**